.github/workflows/test-e2e-desktop.yaml

```diff
@@ -0,0 +1,109 @@
+name: Test E2E Desktop Client
+
+on:
+  pull_request:
+    types: [labeled]
+  workflow_dispatch:
+
+jobs:
+  test:
+    if: ${{ github.repository == 'hashicorp/boundary-ui' && github.event.label.name == 'desktop' }}
+    name: Run Desktop E2E Tests
+    runs-on: ubuntu-22.04
+    env:
+      ENOS_VAR_boundary_edition: "ent"
+      ENOS_VAR_e2e_debug_no_run: true
+      ENOS_VAR_boundary_license_path: "./license.hclic"
+      SETUP_CLI: true
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Node
+        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+        with:
+          node-version: 20
+          cache: "yarn"
+          cache-dependency-path: |
+            yarn.lock
+            ui/desktop/electron-app/yarn.lock
+
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@a1502cd9e758c50496cc9ac5308c4843bcd56d36
+
+      - name: Set up Enos
+        uses: hashicorp/action-setup-enos@b9fa53484a1e8fdcc7b02a118bcf01d65b9414c9
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Vault CLI
+        run: |
+          wget https://releases.hashicorp.com/vault/1.12.2/vault_1.12.2_linux_amd64.zip -O /tmp/vault.zip
+          unzip /tmp/vault.zip -d /usr/local/bin
+
+      - name: Checkout boundary-enterprise
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          repository: hashicorp/boundary-enterprise
+          #          ref: ${{ github.event.inputs.boundary-enterprise-branch }}
+          path: support/src/boundary-enterprise
+          token: ${{ secrets.ELEVATED_GITHUB_TOKEN }}
+
+      - name: GH fix for localhost resolution
+        run: |
+          cat /etc/hosts && echo "-----------"
+          sudo sed -i 's/::1 *localhost ip6-localhost ip6-loopback/::1 ip6 -localhost ip6-loopback/g' /etc/hosts
+          cat /etc/hosts
+          ssh -V
+
+          echo "127.0.0.1 localhost boundary" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 localhost worker" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 localhost vault" | sudo tee -a /etc/hosts
+
+      - name: Get Boundary docker image
+        run: |
+          docker pull hashicorp/boundary-enterprise:latest
+          docker save hashicorp/boundary-enterprise:latest | gzip > /tmp/boundary-enterprise.zip
```

Comment:
We'll need to figure out what version of boundary to run against. If `main`, we'll need to figure out how get the latest docker image

(we'll also need to update the environment variables set in the `set up test infra` step

Reply:
Additionally, there are two version of boundary being specified here
- one for the docker image
- one for the code base we checkout that contains the enos/infra setup code

Do we want to keep those in sync or have them be defined separately? We may also want to allow a user to specify, at a minimum, the version of the docker image (could be useful, say, to run against an LLB for a feature or for release testing).

Reply:
Yeah I think we should keep them in sync. I was actually thinking maybe we should just combine them by checking out the repo for enos and then also building the image ourselves at that version rather than pulling from anywhere.

Reply:
👍 Worth trying

Reply:
It seems building the ent image also will require some adjustments so for now I added a todo and just keep using `latest`